tests/test_function_set.py

```diff
@@ -1,10 +1,63 @@
+import numpy as np
 import cPickle as pickle
 from unittest import TestCase
-from chainer import cuda, FunctionSet
+from chainer import cuda, FunctionSet, Function
 from chainer.functions import Linear
 
 cuda.init()
 
+class MockFunction(Function):
+    def __init__(self, shape):
+        self.p = np.zeros(shape).astype(np.float32)
+        self.gp = np.ones(shape).astype(np.float32)
+
+    @property
+    def parameter_names(self):
+        return ('p', )
+
+    @property
+    def gradient_names(self):
+        return ('gp', )
+
+class TestNestedFunctionSet(TestCase):
+    def setUp(self):
+        self.fs1 = FunctionSet(
+            a = MockFunction((1, 2)))
+        self.fs2 = FunctionSet(
+            fs1 = self.fs1,
+            b  = MockFunction((3, 4)))
+
+    def test_get_sorted_funcs(self):
+        self.assertItemsEqual([k for (k, v) in self.fs2._get_sorted_funcs()], ('b', 'fs1'))
```

Comment:
Use `keys()` instead of the list comprehension: `self.fs2._get_sorted_funcs().keys()`